src/query/xasl_cache.c

```diff
@@ -1803,10 +1828,12 @@ xcache_invalidate_entries (THREAD_ENTRY * thread_p, bool (*invalidate_check) (XA
 		   */
 		  while (xcache_entry->n_cache_clones > 0)
 		    {
-		      xcache_clone_decache (thread_p, &xcache_entry->cache_clones[--xcache_entry->n_cache_clones]);
+		      xcache_clone_decache (thread_p, &xcache_entry->cache_clones[--xcache_entry->n_cache_clones],
+					    xcache_entry);
 		    }
 		  delete_xids[n_delete_xids++] = xcache_entry->xasl_id;
 		}
+	      ATOMIC_INC_32 (&xcache_Memory_usage_cache, -xcache_entry_get_entrysize (xcache_entry));
```

Comment:
```suggestion
	      /* The entry is actually deleted in xcache_entry_uninit(), and may not be deleted immediately. 
	       * The reason for decrement here is to avoid unnecessary cleanup.
	       */
	      ATOMIC_INC_32 (&xcache_Memory_usage_cache, -xcache_entry_get_entrysize (xcache_entry));
```